contracts/test/auction/fa2-auction.test.ts

```diff
@@ -0,0 +1,173 @@
+  let bid_tokens_alice : MintFtParam;
+  let create_ft_param : TokenMetadata;
+
+  beforeAll(async () => {
+    tezos = await bootstrap();
+    empty_metadata_map = new MichelsonMap();
+    tokenIdNft = new BigNumber(1);
+    tokenIdBidToken = new BigNumber(0);
+    bobAddress = await tezos.bob.signer.publicKeyHash();
+    aliceAddress = await tezos.alice.signer.publicKeyHash();
+    nft = {
+      token_metadata: {
+        token_id: tokenIdNft,
+        token_info: empty_metadata_map
+      },
+      owner: bobAddress
+    };
+    bid_tokens_bob = {
+        token_id: tokenIdBidToken,
+        owner: bobAddress,
+        amount : new BigNumber(300)
+    };
+    bid_tokens_alice = {
+        token_id: tokenIdBidToken,
+        owner: aliceAddress,
+        amount : new BigNumber(300)
+    };
+    create_ft_param = {
+        token_id : tokenIdBidToken,
+        token_info : empty_metadata_map
+    };
+  });
+
+  beforeEach(async() =>{
+    $log.info('originating nft faucet...');
+    nftContract = await originateNftFaucet(tezos.bob, bobAddress);
+
+    $log.info('originating ft faucet...');
+    ftContract = await originateFtFaucet(tezos.bob, bobAddress);
+    
+    $log.info('minting nft')
+    const opMintNft = await nftContract.methods.mint([nft]).send();
+    await opMintNft.confirmation();
+    $log.info(`Minted nft. Consumed gas: ${opMintNft.consumedGas}`);
+
+    $log.info('creating fa2 for bids')
+    const opCreateFA2 = await ftContract.methods.create_token(tokenIdBidToken, empty_metadata_map).send();
+    await opCreateFA2.confirmation();
+    $log.info(`Created FA2 Consumed gas: ${opCreateFA2.consumedGas}`);
+
+    $log.info('minting fa2 for bids')
+    const opMintFA2 = await ftContract.methods.mint_tokens([bid_tokens_alice, bid_tokens_bob]).send();
+    await opMintFA2.confirmation();
+    $log.info(`Minted FA2. Consumed gas: ${opMintFA2.consumedGas}`);
+
+    $log.info('originating nft auction with payment in FA2...');
+    nftAuction = await originateEnglishAuctionFA2(tezos.bob, ftContract.address, tokenIdBidToken);
+    nftAuctionBob = await tezos.bob.contract.at(nftAuction.address);
+    nftAuctionAlice = await tezos.alice.contract.at(nftAuction.address);
+
+
+    $log.info('adding auction contract as operator for nft');
+    await addOperator(nftContract.address, tezos.bob, nftAuction.address, tokenIdNft);
+    $log.info('Auction contract added as operator for nft');
+
+    $log.info('adding auction contract as operator for bid token for alice');
+    await addOperator(ftContract.address, tezos.alice, nftAuction.address, tokenIdBidToken);
+    $log.info('Auction contract added as operator for bid token');
+
+    $log.info('adding auction contract as operator for bid token for bob');
+    await addOperator(ftContract.address, tezos.bob, nftAuction.address, tokenIdBidToken);
+    $log.info('Auction contract added as operator for bid token for bob');
+    
+    const fa2_tokens : Fa2_tokens = {
```

Comment:
should the constant and type be a singular form? I. e. `Fa2_token`

Reply:
For these tests that is fine because we only are minting/auctioning a single NFT. 
In the contract code, `amount` can be `> 1`

Reply:
Maybe it will be clearer to just assume amount should be = to 1, given that this contract should be used for NFTs and rename the contract to singular form as you suggested though